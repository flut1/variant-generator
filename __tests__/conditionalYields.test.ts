import createVariants from '../src/index';

describe('createVariants()', () => {
  test('handles conditional yields', () => {
    const result = createVariants(function*() {
      const fruit = yield ['banana', 'apple', 'strawberry'];
      let size = 'normal';

      switch (fruit) {
        case 'banana':
          size = yield ['small', 'normal', 'large'];
          break;
        case 'apple':
          size = yield ['small', 'normal'];
          break;
      }
      let color: string = 'red';

      switch (fruit) {
        case 'banana':
          color = yield ['yellow', 'green'];
          break;
        case 'apple':
          color = yield ['green', 'red'];
          break;
      }

      return {
        fruit,
        size,
        color,
      };
    });

    expect(Array.from(result)).toMatchSnapshot();
  });
});
