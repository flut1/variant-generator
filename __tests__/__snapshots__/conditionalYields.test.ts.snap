// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`createVariants() handles conditional yields 1`] = `
Array [
  Object {
    "color": "yellow",
    "fruit": "banana",
    "size": "small",
  },
  Object {
    "color": "green",
    "fruit": "banana",
    "size": "small",
  },
  Object {
    "color": "yellow",
    "fruit": "banana",
    "size": "normal",
  },
  Object {
    "color": "green",
    "fruit": "banana",
    "size": "normal",
  },
  Object {
    "color": "yellow",
    "fruit": "banana",
    "size": "large",
  },
  Object {
    "color": "green",
    "fruit": "banana",
    "size": "large",
  },
  Object {
    "color": "green",
    "fruit": "apple",
    "size": "small",
  },
  Object {
    "color": "red",
    "fruit": "apple",
    "size": "small",
  },
  Object {
    "color": "green",
    "fruit": "apple",
    "size": "normal",
  },
  Object {
    "color": "red",
    "fruit": "apple",
    "size": "normal",
  },
  Object {
    "color": "red",
    "fruit": "strawberry",
    "size": "normal",
  },
]
`;
